### ISSUE TITLE
Eloquent relationship batching

### ISSUE THREAD

**Issue by KevinAsher** [Folkloreatelier/laravel-graphql#294]
Hi,

I'm working on a generic batching utility for Eloquent Models with Graphql, [here](https://github.com/KevinAsher/DataLoaderTraitExample/blob/master/app/Traits/DataLoaderTrait.php) is the repo with an initial implementation, check out [Types directory](https://github.com/KevinAsher/DataLoaderTraitExample/tree/master/app/GraphQL/Type) for examples in usage. It uses [this](https://github.com/overblog/dataloader-php) dataloader for batching. It would be nice to have some feedback from the community, and perhaps we can include this as an optional add-on to this library.

**Comment by SonarSoftware** [comment]
This is great, I have built something internally to use (as I don't use Eloquent for this API) and it is a crucial part of building a scalable GraphQL API. I didn't dig through all your code to see if you're using it, but the library this is based on has some built in methods to achieve the end goal - see http://webonyx.github.io/graphql-php/data-fetching/#solving-n1-problem

I'm using the 'Deferred' method to load all the root IDs into a query buffer/data loader, and then loading up all the items and providing them back out via the root ID when queried.

**Comment by KevinAsher** [comment]
Yes, the implementation that you mentioned about the buffered IDs is exactly what the Dataloader does, but instead it uses a promise-like approach which follows the original DataLoader created by Facebook.
